### PR TITLE
Collections

### PR DESCRIPTION
**Pull request by jessopb**

### CODE REVIEW

File: src/redux/actions/claims.js [hunk 1]
```diff
@@ -667,6 +705,112 @@ export function doRepost(options: StreamRepostOptions) {
...
+          },
+        });
+        dispatch({
+          type: ACTIONS.UPDATE_PENDING_CLAIMS,
+          data: {
+            claims: [collectionClaim],
+          },
+        });
+        // this only asks for streams and reposts right now
+        dispatch(doFetchClaimListMine(1, 10));
```
Comment (seanyesmunt) [inline src/redux/actions/claims.js:743]: doFetchCollectionListMine

File: src/redux/reducers/collections.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+const defaultState: CollectionState = {
+  builtin: {
+    watchlater: {
+      items: [
+        {
+          url: 'lbry://@seriously#5/seriouspublish#c',
+          claimId: 'c1b740eb88f96b465f65e5f1542564539df1c62e',
+        },
+      ],
+      id: 'watchlater',
```
Comment (seanyesmunt) [inline src/redux/reducers/collections.js:18]: make watchlater a constant

File: src/redux/reducers/collections.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        lists,
+      });
+    },
+
+    [ACTIONS.PLAYLIST_UPDATE]: (state, action) => {
+      const { unpublished: lists } = state;
+      const newLists = Object.assign({}, lists);
+      const { id, playlist } = action.data;
+      newLists[id] = playlist;
+      newLists[id]['updatedAt'] = getTimestamp();
```
Comment (seanyesmunt) [inline src/redux/reducers/collections.js:81]: move this to the action

File: src/redux/reducers/collections.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  unpublished: {},
+  saved: [],
+  mine: [],
+  isResolvingCollectionById: [],
+  error: null,
+};
+
+const collectionsReducer = handleActions(
+  {
+    [ACTIONS.PLAYLIST_CREATE]: (state, action) => {
```
Comment (seanyesmunt) [inline src/redux/reducers/collections.js:46]: COLLECTION_UNPUBLISHED_CREATE

File: src/redux/reducers/collections.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        error: action.data.message,
+      });
+    },
+
+    [ACTIONS.COLLECTION_RESOLVE_STARTED]: (state, action) => {
+      const { ids } = action.data;
+      const { isResolvingCollectionById } = state;
+      const newResolving = isResolvingCollectionById.concat();
+      ids.forEach(id => {
+        newResolving.push(id);
```
Comment (seanyesmunt) [inline src/redux/reducers/collections.js:99]: resolvingCollectionById[id] = true

From comment above about keeping this the same type as isResolvingClaimsById

File: src/redux/selectors/claims.js [hunk 6]
```diff
@@ -543,6 +554,13 @@ export const selectMyChannelUrls = createSelector(
   claims => (claims ? claims.map(claim => claim.canonical_url || claim.permanent_url) : undefined)
 );
 
+export const selectMyCollectionIds = createSelector(
+  selectState,
+  state => state.myCollectionClaims
+);
+
+// selectMyCollectionUrls
```
Comment (seanyesmunt) [inline src/redux/selectors/claims.js:562]: comment

File: src/redux/selectors/collections.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+import { createSelector } from 'reselect';
+import { selectMyCollectionIds } from 'redux/selectors/claims';
+
+const selectState = (state: { collections: CollectionState }) => state.collections;
+
+export const selectSavedCollectionIds = createSelector(
+  selectState,
+  collectionState => collectionState.saved
+);
+export const selectBuiltinPlaylists = createSelector(
```
Comment (seanyesmunt) [inline src/redux/selectors/collections.js:11]: selectBuiltinCollections

File: src/redux/selectors/collections.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    );
+    return mySavedPlaylists;
+  }
+);
+
+export const makeSelectIsResolvingCollectionForId = (id: string) =>
+  createSelector(
+    selectState,
+    state => {
+      return state.isResolvingCollectionById.includes(id);
```
Comment (seanyesmunt) [inline src/redux/selectors/collections.js:49]: return state.isResolvingCollectionById[id]

**Review by seanyesmunt — CHANGES_REQUESTED** [review]

File: src/redux/actions/collections.js [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    }
+    return val;
+  }
+
+  const promises = [];
+  collectionIds.forEach(collectionId => {
+    promises.push(resolveCollection(collectionId));
+  });
+
+  const resolvedCollectionItemsById: Array<{ [string]: Array<GenericClaim> }> = await Promise.all(
```
[the remark this answers is not in the document]
Comment (jessopb) [reply src/redux/actions/collections.js:60]: Array/< { claimId: string, items: ?Array<GenericClaim> } >

File: dist/flow-typed/Collections.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
+declare type Collection = {
+  id: string,
+  items: Array<?string>,
+  name: string,
+  type: string,
+  updatedAt: number,
+  totalItems?: number,
+  sourceid?: string, // if copied, claimId of original collection
```
Comment (infinite-persistence) [inline dist/flow-typed/Collections.js:8]: typo?

File: src/redux/actions/claims.js [hunk 11]
```diff
@@ -11,13 +11,24 @@ import {
...
 } from 'redux/selectors/claims';
+
 import { doFetchTxoPage } from 'redux/actions/wallet';
 import { selectSupportsByOutpoint } from 'redux/selectors/wallet';
 import { creditsToString } from 'util/format-credits';
 import { batchActions } from 'util/batch-actions';
 import { createNormalizedClaimSearchKey } from 'util/claim';
 import { PAGE_SIZE } from 'constants/claim';
+import {
+  makeSelectEditedCollectionForId,
```
Comment (infinite-persistence) [inline src/redux/actions/claims.js:24]: There are a few unused variables in this file.

File: src/redux/actions/collections.js [hunk 12]
```diff
@@ -0,0 +1,489 @@
+// @flow
+import * as ACTIONS from 'constants/action_types';
+import { v4 as uuid } from 'uuid';
+import Lbry from 'lbry';
+import { doClaimSearch, doAbandonClaim } from 'redux/actions/claims';
+import { makeSelectClaimForClaimId } from 'redux/selectors/claims';
+import {
+  makeSelectCollectionForId,
+  // makeSelectPublishedCollectionForId, // for "save" or "copy" action
+  makeSelectMyPublishedCollectionForId,
```
Comment (infinite-persistence) [inline src/redux/actions/collections.js:10]: Unused

**Review by infinite-persistence — COMMENTED** [review]
First pass ok.  Mostly indirectly tested using the Desktop PR.  
Only minor (ignorable) cleanup here.

**Pull request merged**